src/hotspot/share/runtime/threads.cpp

```diff
@@ -1332,8 +1332,11 @@ void Threads::print_on(outputStream* st, bool print_stacks,
           if (p->is_vthread_mounted()) {
             const oop vt = p->vthread();
             assert(vt != nullptr, "vthread should not be null when vthread is mounted");
-            st->print_cr("   Mounted virtual thread \"%s\" #" INT64_FORMAT, JavaThread::name_for(vt), (int64_t)java_lang_Thread::thread_id(vt));
-            p->print_vthread_stack_on(st);
+            if (vt != thread_oop) {
+              // JavaThread._vthread can refer to the carrier thread. Print only if _vthread refers to a virtual thread.
```

Comment:
Nit: please put the comment before the `if`

Reply:
Thanks ! I've fixed that in: 05c0f75acd6a37815146c7a7e1b30aede47ab77a